operator/webhooks/pod_mutating_webhook.go

```diff
@@ -52,19 +62,30 @@ func (a *PodMutatingWebhook) Handle(ctx context.Context, req admission.Request)
 
 	isAnnotated, err := a.isKeptnAnnotated(pod)
 	if err != nil {
+		span.SetStatus(codes.Error, "Invalid annotations")
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 	if isAnnotated {
 		logger.Info("Resource is annotated with Keptn annotations, using Keptn scheduler")
 		pod.Spec.SchedulerName = "keptn-scheduler"
 		logger.Info("Annotations", "annotations", pod.Annotations)
+
+		app, _ := pod.Annotations[common.AppAnnotation]
+		workload, _ := pod.Annotations[common.WorkloadAnnotation]
+		version, _ := pod.Annotations[common.VersionAnnotation]
+		span.SetAttributes(semconv.ApplicationName.String(app))
```

Comment:
If this is needed more often, we could put the span.SetAttributes in a function ...

Reply:
Yes, I was also thinking to have a set of functions in `semconv` that automatically add these attributes to a span: e.g., `setSpanAttributesWithWorkload(Span, Workload)`

Reply:
If we are to introduce a common module that is required by both the scheduler, as well as the operator, I would prefer to have this in a separate repository. Having dependencies to a module in the same repository proved to be quite painful and error prone - e.g. if changes to both the common module and a module that requires it are made in the same PR, the common module needs to be addressed via a pushed commit (i.e. in this case, a commit of the PR). In this scenario, the build would be broken as soon as the branch will be deleted, and a separate PR to update the commit id of the dependency will be needed.